browser/ephemeral_storage/blob_url_browsertest.cc

```diff
@@ -0,0 +1,320 @@
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "content/public/common/url_constants.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/test_navigation_observer.h"
+#include "content/public/test/test_utils.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+#include "url/gurl.h"
+#include "url/origin.h"
+
+namespace {
+
+constexpr char kCreateBlobScript[] = "URL.createObjectURL(new Blob([$1]))";
+
+constexpr char kFetchBlobScript[] = R"(
+(async function() {
+  try {
+    const response = await fetch($1);
+    return await response.text();
+  } catch (err) {
+    return 'error';
+  }
+})();)";
+
+constexpr char kWorkerScript[] = R"(
+  (async() => {
+    try {
+      const response = await fetch($1);
+      postMessage(await response.text());
+    } catch (err) {
+      postMessage('error');
+    }
+  })();
+)";
+
+constexpr char kFetchBlobViaWorkerScript[] = R"(
+  new Promise(async (resolve) => {
+    const blobURL = URL.createObjectURL(new Blob([$1]));
+    const dedicatedWorker = new Worker(blobURL);
+    dedicatedWorker.addEventListener('message', e => {
+      resolve(e.data);
+    });
+  });
+)";
+
+}  // namespace
+
+class BlobUrlBrowserTest : public EphemeralStorageBrowserTest {
+ public:
+  void SetUpOnMainThread() override {
+    EphemeralStorageBrowserTest::SetUpOnMainThread();
+    ASSERT_TRUE(embedded_test_server()->Start());
+  }
+
+  static GURL RegisterBlob(content::RenderFrameHost* render_frame_host,
+                           base::StringPiece content) {
+    return GURL(EvalJs(render_frame_host,
+                       content::JsReplace(kCreateBlobScript, content))
+                    .ExtractString());
+  }
+
+  static content::EvalJsResult FetchBlob(
+      content::RenderFrameHost* render_frame_host,
+      const GURL& url) {
+    auto fetch_result = EvalJs(
+        render_frame_host, content::JsReplace(kFetchBlobScript, url.spec()));
+    auto fetch_via_webworker_result = EvalJs(
+        render_frame_host,
+        content::JsReplace(kFetchBlobViaWorkerScript,
+                           content::JsReplace(kWorkerScript, url.spec())));
+    EXPECT_EQ(fetch_result.value, fetch_via_webworker_result.value);
+    EXPECT_EQ(fetch_result.error, fetch_via_webworker_result.error);
+    return fetch_result;
+  }
+
+  static std::vector<content::RenderFrameHost*> GetFrames(
+      content::RenderFrameHost* main_frame) {
+    return {
+        main_frame,
+        content::ChildFrameAt(main_frame, 0),
+        content::ChildFrameAt(main_frame, 1),
+        content::ChildFrameAt(main_frame, 2),
+        content::ChildFrameAt(main_frame, 3),
+    };
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(BlobUrlBrowserTest, BlobsArePartitioned) {
```

Comment:
we need to test that they are not partitioned when the feature is off and also test based on content settings since we're checking those here https://github.com/brave/brave-core/pull/12686/files#diff-430f56739d7df3701057dcb1c3ea5412094160cc91008cfd596d42d889e4d8d4R72

Reply:
we also need to check that it's disabled when the main ephemeral storage is disabled and check that only 1p is disabled when the 1p flag is disabled